modules/optimization/include/optimizationreporters/ParameterMeshOptimization.h

```diff
@@ -9,18 +9,35 @@
 
 #pragma once
 
+#include "OptimizationDataHelper.h"
 #include "OptimizationReporterBase.h"
+
 class ParameterMesh;
 
 /**
  * Mesh-based parameter optimization
  */
 class ParameterMeshOptimization : public OptimizationReporterBase
 {
+private:
+  /// Data helper for generating objective value.
+  OptimizationDataHelper _opt_data;
+
 public:
   static InputParameters validParams();
   ParameterMeshOptimization(const InputParameters & parameters);
 
+  void execute() override;
+
+  virtual Real computeObjective() override;
+
+  /// measurement values
```

Comment:
Should we move _measurement_values _simulation_values _misfit_values to the base class since the derived classes all need them?  Or you creating new classes that don't need this data?  If so, we should make invOptReporterBase that has this data.